core/java/iesi-core/src/main/java/io/metadew/iesi/script/action/fwk/FwkSetEnvironment.java

```diff
@@ -41,9 +45,17 @@ public void prepare() {
 
     protected boolean executeAction() throws InterruptedException {
         String environmentName = convertEnvironmentName(getEnvironmentName().getValue());
-        this.getExecutionControl().setEnvironment(getActionExecution(), environmentName);
-        this.getActionExecution().getActionControl().increaseSuccessCount();
-        return true;
+
+        //Check if environment exists
+        Optional<Environment> environment = EnvironmentConfiguration.getInstance().get(new EnvironmentKey(environmentName));
+        if (environment.equals(Optional.empty())) {
```

Comment:
Replace with environment.isPresent()

Reply:
updated in https://github.com/metadew/iesi/pull/28/commits/a03bed176e393ad556ff7bdb162dc85f2bd188d2